app/backend/src/couchers/servicers/search.py

```diff
@@ -660,10 +673,13 @@ def EventSearch(self, request, context, session):
                 EventOccurrence.start_time.desc()
             )
 
-        statement = statement.limit(page_size + 1)
+        total_items = session.execute(select(func.count()).select_from(statement.subquery())).scalar()
+        # Apply pagination by page number
+        statement = statement.offset(offset).limit(page_size) if request.page_number else statement.limit(page_size + 1)
```

Comment:
It seems `offset` is not defined if you have `pagination_type == "page_token"`

Reply:
I changed it back, turns out I did not need these changes.